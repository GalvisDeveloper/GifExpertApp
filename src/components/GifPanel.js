

import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridPanel } from './GifGridPanel';
import PropTypes from 'prop-types';

const GifPanel = ({ category }) => {

    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3 className="animate__animated animate__bounce animate__rotateIn">{category}</h3>
            { loading && <p className="animate__animated animate__bounce animate__flash">Loading...</p>}
            <div className="card-grid">
                {
                    images.map(img => (
                        <GifGridPanel
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}

GifPanel.propTypes = {
    category: PropTypes.string.isRequired,
}

export default GifPanel;