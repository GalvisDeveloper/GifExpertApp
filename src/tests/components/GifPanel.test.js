
import React from 'react';
import { shallow } from 'enzyme';
import GifPanel from '../../components/GifPanel';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el componente <GifPanel />', () => {

    const category = 'Moto gp';

    test('Prueba 1: Coincidir el snapshot ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow(<GifPanel category={category} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Prueba 2: Mostrar items cuando se cargan las imagenes useFetchGifs(customHook)', () => {

        const gifs = [{
            id: '01',
            url: 'https://www.google.com/',
            title: 'Google'
        },
        {
            id: '02',
            url: 'https://www.youtube.com/',
            title: 'Youtube'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifPanel category={category} />);

        // expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridPanel').length).toBe(gifs.length);
    })


});
