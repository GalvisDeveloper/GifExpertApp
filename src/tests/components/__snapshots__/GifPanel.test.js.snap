// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifPanel /> Prueba 1: Coincidir el snapshot  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce animate__rotateIn"
  >
    Moto gp
  </h3>
  <p
    className="animate__animated animate__bounce animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el componente <GifPanel /> Prueba 2: Mostrar items cuando se cargan las imagenes useFetchGifs(customHook) 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce animate__rotateIn"
  >
    Moto gp
  </h3>
  <div
    className="card-grid"
  >
    <GifGridPanel
      id="01"
      key="01"
      title="Google"
      url="https://www.google.com/"
    />
  </div>
</Fragment>
`;
